Add structured data fields to Spree stores

rahulsingh321 opened the pull request:
## Summary
This Commit provides the possibility to store legal information in the
store resource. Given the wide array of different information required
by country, the information stored here provides a baseline for most
countries, but might need to be augmented in some jurisdictions. This
commit is not meant as legal advisory. Inform yourself about eventual
requirements in your country of operation.

Some additional information that might be needed:
 - Legal Representative,
 - Share Capital of the company,
 - Contacts for privacy.

Schema.org has a schema to markup organizations that has been widely
adopted by various search engines:

|        | https://schema.org/Organization Support / Docs                                    |
|--------|-----------------------------------------------------------------------------------|
| Bing   | https://www.bing.com/webmasters/help/?topicid=cc507c09                            |
| Google | https://developers.google.com/search/docs/appearance/structured-data/organization |
| Yandex | https://yandex.com/support/webmaster/schema-org/what-is-schema-org.html           |


This update allows to return all organization data on frontend via jsonb
if implemented according to schema documentation on a per store basis
and edit the data via API and Backend (Old / New).

The implemented fields are:

- legal_name,
- contact_email,
- contact_phone,
- vat_id,
- tax_id,
- address1,
- address2,
- Zip code and city,
- state_name,
- country_id,
- and state_id.

Some additional notes regarding the size of the commit:
While this commit seems massive in size, no functionality has been added
apart from storing the values, no logical changes have been made, hence
the commit while bigger in size is simple to analyse. Apart from testing
routines, this commit does not contain functional code. Where possible
the structure of current address forms has been adopted.

Fixes https://github.com/solidusio/solidus/issues/6173

## Screenshot
![Screenshot from 2025-03-02 17-48-07](https://github.com/user-attachments/assets/5a954b34-9258-4418-8697-aafb86840fc1)
![Screenshot from 2025-03-02 17-47-52](https://github.com/user-attachments/assets/2616e94f-a29c-4ad3-b9bd-9d233c570953)
![Screenshot from 2025-03-02 17-47-35](https://github.com/user-attachments/assets/c20cb017-62a5-45a7-be3a-bda156e6f5c2)
![Sample-Store-Stores-03-04-2025_10_02_PM](https://github.com/user-attachments/assets/288a1d9a-777e-403f-a3a9-26434153454c)



<!--
  Please include a summary of your changes, along with any useful context.
  Your contribution will be merged under the terms of the license of the parent repository (usually a FreeBSD License).

  You're encouraged to include screenshots in case of visual changes.

  If needed, you can reference other PRs or issues here with #ISSUE-NUMBER.
  You can use GitHub-specific syntax, e.g.

  Fixes #ISSUE-NUMBER

  However, if you do not have merge permissions on the repo, issues won't be auto-closed.
-->

## Checklist

Check out our [PR guidelines](https://github.com/solidusio/.github/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [ ] [I agree that my PR will be published under the same license as Solidus](https://github.com/solidusio/solidus/blob/main/LICENSE.md).
- [ ] I have written a thorough PR description.
- [ ] I have kept my commits small and atomic.
- [ ] [I have used clear, explanatory commit messages](https://github.com/solidusio/.github/blob/main/CONTRIBUTING.md#writing-good-commit-messages).

The following are not always needed:

- 📖 I have updated the README to account for my changes.
- 📑 I have documented new code [with YARD](https://www.rubydoc.info/gems/yard/file/docs/Tags.md).
- 🛣️ I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).
- ✅ I have added automated tests to cover my changes.
- 📸 I have attached screenshots to demo visual changes.

fthobe commented:
1. Please restructure the form order as in attached image
2. contact email is missing on forms
3. Save button should be at right top and right bottom as in figma (https://www.figma.com/design/h7eZRxh8KhPakTMr7c9NlB/Solidus-%E2%80%94-Admin-Dashboard?node-id=419-7816&p=f&t=u5VE4L1wTTlfzrkw-0)
4. Make separate cards according to mock up for old backend

<img width="177" alt="Screenshot 2025-02-27 at 19 59 39" src="https://github.com/user-attachments/assets/bbb02216-978b-407f-b308-c94bb33a6fbb" />

rahulsingh321 committed:
[Compliance / SEO] Add structured data fields to Spree stores

This Commit provides the possibility to store legal information in the
store resource. Given the wide array of different information required
by country, the information stored here provides a baseline for most
countries, but might need to be augmented in some jurisdictions. This
commit is not meant as legal advisory. Inform yourself about eventual
requirements in your country of operation.

Some additional information that might be needed:
 - Legal Representative,
 - Share Capital of the company,
 - Contacts for privacy.

Schema.org has a schema to markup organizations that has been widely
adopted by various search engines:

|        | https://schema.org/Organization Support / Docs
|
|--------|-----------------------------------------------------------------------------------|
| Bing   | https://www.bing.com/webmasters/help/?topicid=cc507c09
|
| Google |
https://developers.google.com/search/docs/appearance/structured-data/organization
|
| Yandex |
https://yandex.com/support/webmaster/schema-org/what-is-schema-org.html
|

This update allows to return all organization data on frontend via jsonb
if implemented according to schema documentation on a per store basis
and edit the data via API and Backend (Old / New).

The implemented fields are:

- legal_name,
- contact_email,
- contact_phone,
- vat_id,
- tax_id,
- address1,
- address2,
- Zip code and city,
- state_name,
- country_id,
- and state_id.

Some additional notes regarding the size of the commit:
While this commit seems massive in size, no functionality has been added
apart from storing the values, no logical changes have been made, hence
the commit while bigger in size is simple to analyse. Apart from testing
routines, this commit does not contain functional code. Where possible
the structure of current address forms has been adopted.

Fixes #6173

tvdeyen commented:
@fthobe please stop framing PRs and issues "your way" in the PR title. Thanks.

tvdeyen commented on admin/app/components/solidus_admin/stores/address_form/component.js:
```diff
@@ -0,0 +1,56 @@
+import { Controller } from "@hotwired/stimulus"
```
We already have a address form component that handles states. Can we use that one instead of implementing a second one?

fthobe replied:
Yes, we can look into reusing the logic to pick a state. Does it make sense if everything else should be detached from the address table?

tvdeyen commented on admin/app/components/solidus_admin/stores/edit/component.html.erb:
```diff
@@ -0,0 +1,87 @@
+  <%= form_for @store, url: solidus_admin.store_path(@store), html: { id: form_id } do |f| %>
+    <%= page_with_sidebar do %>
+      <%= page_with_sidebar_main do %>
+        <%= render component("ui/panel").new(title: t(".store_settings")) do %>
```
Why haven't you used the existing address form component?

fthobe replied:
It didn't seem indicated to invoke the address list for performance reasons. 
These values should be cached and provided separately as otherwise the address tabled needs to be used for values that in some scenarios could be embedded on every page depending on the implementation.

tvdeyen reviewed (changes requested):
We should use the existing address form component, instead of introducing a second one. Also the PR adds a general new admin form for managing stores, which is very much appreciated, but this should at least be a dedicated commit the extra builds on top of and should no be combined into one commit, otherwise it is hard to see what the actual change (from your PR title) is.

fthobe commented:
@tvdeyen replied to your comments

tvdeyen commented:
> > @fthobe please stop framing PRs and issues "your way" in the PR title. Thanks.
> 
> @tvdeyen could you suggest a better name to get a hang of it? The resource is stores and the fields we add are the structured data fields. Please advise, not being condescending, trying to learn.

The name is ok, the "[Tag]"-Part is unnecessary

fthobe commented:
@tvdeyen I worked with the guys of schema to adjust the [Organization tags a little and complete them for European entities](https://github.com/schemaorg/schemaorg/pull/4387) as well and it no objection is raised that passes next week. 


Was there anything else to complain about other than the address form? 

Given the previous discussions the address list for me is a dead cow and I would like to store this in simple columns. Is that ok for you?

fthobe commented:
@tvdeyen do I close this or leave it open?
It provides full imprint / impressum via the store resource. I extended the structured data model on schema as well.

jarednorman commented:
This is a 1000+ line change that no longer has someone to champion it. Not a bad idea, but I'm going to close due to maintenance risk. I'm by no means opposed to revisiting this if it's a feature someone needs/wants.
